pandas/core/generic.py

```diff
@@ -4688,6 +4688,31 @@ def _where(self, cond, other=np.nan, inplace=False, axis=None, level=None,
         Returns
         -------
         wh : same type as caller
+
+        Notes
+        -----
+        The signature for :func:`DataFrame.where` differs from
+        :func:`numpy.where`. Roughly ``df1.where(m, df2)`` is equivalent to
+        ``np.where(m, df1, df2)``.
+
+        Examples
+        --------
+        >>> df = pd.DataFrame(np.arange(10).reshape(-1, 2), columns=['A', 'B'])
```

Comment:
start off by showing an example where other is ``np.nan`` (e.g. the default).

also put in a referene to where in the docs: http://pandas.pydata.org/pandas-docs/stable/indexing.html#the-where-method-and-masking; you can also use some of those examples if that helps.

Also if you can add an example in the docs themselves (in a ::note section that shows how it is like np.where)